tfhe/docs/how_to/rayon_crate.md

```diff
@@ -0,0 +1,180 @@
+# Making rayon and tfhe-rs work together
+
+[rayon](https://crates.io/crates/rayon) is a popular create to easily write multi-threaded code in Rust.
+
+It is possible to use rayon to write multi-threaded tfhe-rs code. However due to internal details of `rayon` and
+`tfhe-rs`, there is some special setup that needs to be done.
+
+## Single Client Application
+
+### The problem 
+
+The high level api requires to call `set_server_key` on each thread where computations needs to be done.
+So a first attempt at using rayon with tfhe-rs might look like this:
+
+```rust
+use rayon::prelude::*;
```

Comment:
use a capital R otherwise this code may crash when run as a doctest ?

Reply:
yep, just asking if it's worth the risk

Reply:
I'm confident it wont

This program spaws two task

thread1 is going to do the first one, and adverstive the second as being stealable,
thread2 may come in and steal the task.

If the task is stolen, when thread1 finished its initial task, is would have another task to steal, and just wait for thread 2
if thread1 had a task that did a call to `set_server_key` to steal, then the problem would appear

Reply:
ok !